src/Microsoft.AspNetCore.Routing/Matching/ParameterPolicyFactory.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Routing.Patterns;
+
+namespace Microsoft.AspNetCore.Routing.Matching
+{
+    internal abstract class ParameterPolicyFactory
+    {
+        public abstract IParameterPolicy Create(string parameterName, string value, bool optional);
+
+        public abstract IParameterPolicy Create(string parameterName, IRouteConstraint value, bool optional);
```

Comment:
Is there any real reason we need this overload along with the `IParameterPolicy` overload?